### PR TITLE
Support multiline placeholder and autolayout

### PR DESCRIPTION
**Pull request by kouheiszk**
- Support multiline placeholder
- Support autolayout on storyboard

### CODE REVIEW

**Comment by devxoul** [comment]
Thanks for your work, @kouheiszk!

But I can't merge this PR right now, because it cause misalignment of placeholder label when `self.textField.textAlignment` is set to `NSTextAlignmentCenter`.

ios simulator screen shot jan 27 2015 2 22 22 pm

Thanks.

**Comment by kouheiszk** [comment]
@devxoul I updated pull request to fix misalignment :D

File: UITextView+Placeholder/UITextView+Placeholder.m [hunk 1]
```diff
@@ -144,16 +145,18 @@ - (void)updatePlaceholderLabel {
...
     self.placeholderLabel.font = self.font;
     self.placeholderLabel.textAlignment = self.textAlignment;
-    [self.placeholderLabel sizeToFit];
 
     CGRect frame = self.placeholderLabel.frame;
     frame.origin.x = self.textContainer.lineFragmentPadding + self.textContainerInset.left;
     frame.origin.y = self.textContainerInset.top;
     frame.size.width = (CGRectGetWidth(self.bounds) - frame.origin.x -
                         self.textContainer.lineFragmentPadding - self.textContainerInset.right);
+    frame.size.height = [self.placeholderLabel sizeThatFits:frame.size].height;
```
Comment (devxoul) [inline UITextView+Placeholder/UITextView+Placeholder.m]: At line 148 and 158, `sizeThatFits` called twice. I think we can use `sizeToFit` just once at the end of setting `placeholderLabel.font` and `textAlignment`. (original code)

Is this intended?

**Comment by devxoul** [comment]
What does "Support autolayout on storyboard" means? Your code looks just migrating `textView` from code to storyboard.

Auto layout seemed to be work without any changes (on branch master):

```objc
UITextView *textView = [[UITextView alloc] init];
textView.placeholder = @"How are you?";
textView.font = [UIFont systemFontOfSize:15];
[self.view addSubview:textView];

textView.translatesAutoresizingMaskIntoConstraints = NO;
[self.view addConstraint:[NSLayoutConstraint constraintWithItem:textView
                                                     attribute:NSLayoutAttributeLeft
                                                     relatedBy:NSLayoutRelationEqual
                                                        toItem:self.view
                                                     attribute:NSLayoutAttributeLeft
                                                    multiplier:1
                                                       constant:10]];
[self.view addConstraint:[NSLayoutConstraint constraintWithItem:textView
                                                      attribute:NSLayoutAttributeRight
                                                      relatedBy:NSLayoutRelationEqual
                                                         toItem:self.view
                                                      attribute:NSLayoutAttributeRight
                                                     multiplier:1
                                                       constant:-10]];
[self.view addConstraint:[NSLayoutConstraint constraintWithItem:textView
                                                      attribute:NSLayoutAttributeTop
                                                      relatedBy:NSLayoutRelationEqual
                                                         toItem:self.view
                                                      attribute:NSLayoutAttributeTop
                                                     multiplier:1
                                                       constant:100]];
[self.view addConstraint:[NSLayoutConstraint constraintWithItem:textView
                                                      attribute:NSLayoutAttributeHeight
                                                      relatedBy:NSLayoutRelationEqual
                                                         toItem:self.view
                                                      attribute:NSLayoutAttributeHeight
                                                     multiplier:1
                                                       constant:0]];

```

Is there any problem with auto layout I didn't catch?

**Comment by devxoul** [comment]
@kouheiszk, adding multiline support merged into master at #3. If there are any issue with auto layout, please notify me. Thanks.

**Pull request closed without merging**